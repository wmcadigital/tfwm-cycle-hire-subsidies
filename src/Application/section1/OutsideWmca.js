import { useLocation, useNavigate } from "react-router-dom";

import Header from "../../common/Header";
import BreadCrumb from "../../common/BreadCrumb";
import Question from "../../common/Question";
import ButtonLink from "../../common/ButtonLink";

const OutsideWmca = () => {
  const location = useLocation();
  const navigate = useNavigate();

  return (
    <>
      <Header heading="Apply for support with West Midlands Cycle Hire costs" />
      <main className="wmnds-container wmnds-container--main wmnds-p-b-lg wmnds-grid">
        <div className="wmnds-col-1 wmnds-col-md-2-3">
          <BreadCrumb currentPageName="Apply for free cycle hire" />
          <div className="wmnds-col-1 wmnds-m-b-md">
            <ButtonLink
              callback={() =>
                navigate("/", {
                  state: location.state,
                  replace: true,
                })
              }
            >
              {`< Back`}
            </ButtonLink>
          </div>
          <Question text="Sorry, your postcode is not within our eligible area." />
          <p>
            {`To be eligible for this service you must live in the `}
            West Midlands Combined Authority area.
          </p>
        </div>
      </main>
    </>
  );
};

export default OutsideWmca;
