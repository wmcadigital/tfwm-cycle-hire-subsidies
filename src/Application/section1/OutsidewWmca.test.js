import { BrowserRouter } from "react-router-dom";
import { createRenderer } from "react-test-renderer/shallow";
import { screen, render, fireEvent } from "@testing-library/react";

import OutsideWmca from "./OutsideWmca";

const mockNavigate = jest.fn();

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useLocation: () => ({
    pathname: "/",
    state: { formValues: { name: "Test" } },
  }),
  useNavigate: () => mockNavigate,
}));

describe("OutsideWmca", () => {
  it("renders the page as expected", () => {
    const renderer = createRenderer();

    renderer.render(
      <BrowserRouter>
        <OutsideWmca />
      </BrowserRouter>
    );

    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

  it("clicking back will navigate to registration form with form state", () => {
    render(
      <BrowserRouter>
        <OutsideWmca />
      </BrowserRouter>
    );

    const backLink = screen.getByText("< Back");

    expect(mockNavigate).toBeCalledTimes(0);

    fireEvent.click(backLink);

    expect(mockNavigate).toBeCalledTimes(1);
    expect(mockNavigate).toBeCalledWith("/registration", {
      state: { formValues: { name: "Test" } },
      replace: true,
    });
  });
});
