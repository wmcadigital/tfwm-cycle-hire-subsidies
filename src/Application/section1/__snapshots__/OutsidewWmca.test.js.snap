// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OutsideWmca renders the page as expected 1`] = `
<Router
  location={
    Object {
      "hash": "",
      "key": "default",
      "pathname": "/",
      "search": "",
      "state": null,
    }
  }
  navigationType="POP"
  navigator={
    Object {
      "action": "POP",
      "back": [Function],
      "block": [Function],
      "createHref": [Function],
      "forward": [Function],
      "go": [Function],
      "listen": [Function],
      "location": Object {
        "hash": "",
        "key": "default",
        "pathname": "/",
        "search": "",
        "state": null,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <OutsideWmca />
</Router>
`;
